prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1116 @@
+                                      .select(Tables.ACTION_LOG.asterisk())
+                                      .from(Tables.ACTION_LOG)
+                                      .fetchInto(ActionLog::class.java)
+
+            assert(actionLogRecords.size >= expectedContentList.size)
+
+            for (expectedActionLogRecordContent in expectedContentList) {
+                // -1 because the indexing is not zero-based and we need to modify it to address the right element
+                // in the actionLogRecords array
+                val actualIndex = expectedActionLogRecordContent.index - 1
+                val actualActionLogRecord = actionLogRecords[actualIndex]
+
+                assertEquals(expectedActionLogRecordContent.reportId, actualActionLogRecord.reportId)
+                assertEquals(expectedActionLogRecordContent.type, actualActionLogRecord.type)
+                assertEquals(expectedActionLogRecordContent.scope, actualActionLogRecord.scope)
+
+                expectedActionLogRecordContent.actionLogDetail?.let {
+                    val expected = expectedActionLogRecordContent.actionLogDetail
+                    expected.javaClass.let { assertEquals(it, actualActionLogRecord.detail.javaClass) }
+                    expected.scope.let { assertEquals(it, actualActionLogRecord.detail.scope) }
+                    expected.message.let { assertEquals(it, actualActionLogRecord.detail.message) }
+                    expected.errorCode.let { assertEquals(it, actualActionLogRecord.detail.errorCode) }
+                }
+            }
+        }
+    }
+
+    @BeforeEach
+    fun beforeEach() {
+        mockkObject(QueueAccess)
+        every { QueueAccess.sendMessage(any(), any()) } returns Unit
+        mockkObject(BlobAccess)
+        every { BlobAccess getProperty "defaultBlobMetadata" } returns getBlobContainerMetadata()
+        mockkObject(BlobAccess.BlobContainerMetadata)
+        every { BlobAccess.BlobContainerMetadata.build(any(), any()) } returns getBlobContainerMetadata()
+        mockkConstructor(DatabaseLookupTableAccess::class)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `should send valid FHIR report only to receiver listening to full-elr`() {
+        // set up
+        val reportContents =
+            listOf(
+                validFHIRRecord1
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        fhirFunctions.doRoute(queueMessage, 1, createFHIRRouter())
```

Comment:
nit: having default values in some of these setup functions makes it harder to see what the actual test setup is. For example, the test below actually configures the org that createFHIRRouter will use, so its easier to see what the actual setup is for that test.